### ISSUE TITLE
Add lambda predicates to "shouldContain" 

### ISSUE THREAD

**Issue by igorwojda** [MarkusAmshove/Kluent#153]
Following kotlin collection quantity predicates, we could add equivalent checks to kluent.

```
    data class User(val name: String)

    val list = listOf(User("Markus"), User("Igor"))
    
    // Kotlin
    list.all { it.name == "Markus" }
    list.none { it.name == "Markus" }
    // single is a bit different returning object not Boolean
    list.single { it.name == "Markus" }
    list.any { it.name == "Markus" }
    

    // kluent
    list shouldContainAll { it.name == "Markus" }
    list shouldContainNone { it.name == "Markus" }
    list shouldContainSingle { it.name == "Markus" }
    list shouldContainAny { it.name == "Markus" }
```

This would be very helpful when working with "complex" POJO classes

BTW probably existing `shouldContainSome` should be deprecated and renamed to `shouldContainAny` to follow kotlin conventions

**Comment by MarkusAmshove** [comment]
This looks like a good addition, also the deprecation of `shouldContainSome` makes sense.

Are you interested in opening a PR? 
You can annotate `shouldContainSome` as deprecated and guide to the new implementation, I will then remove it with 2.0

**Comment by igorwojda** [comment]
I could try to implement this in the upcoming weeks. I would appreciate some guideline beforehand mainly the info which files should be updated and mabye some general tips on the important stuff.

**Comment by sheix** [comment]
Take a look at `should match at least one of` and `should match all with`

**Comment by MarkusAmshove** [comment]
Sure, no problem :-)

Everything about collections is currently in [Collections.kt](https://github.com/MarkusAmshove/Kluent/blob/master/common/src/main/kotlin/org/amshove/kluent/Collections.kt) within the common module, so it's available to all platforms.

This feature is also something that fits in there, because it will most likely work, since it's plain Kotlin.

If you're done with the implementation and tests you can run

`./gradlew build` for the JVM build
`./gradlew build -PJS` for the JavaScript build
`./gradlew build -PNATIVE` for the Native build
`./gradlew build -PANDROID` for the Android build

if you want to make sure that it works on every platform. (there is also a `buildAll.sh`, which builds all platforms)

**Comment by javatarz** [comment]
`BooleanArray.shouldContainSome(expected: BooleanArray)` needed an equivalent and the best I could come up with is `BooleanArray.shouldContainTrue()` and `Boolean.shouldContainFalse()`.

My PR has more details.

**Comment by MarkusAmshove** [comment]
This is now released within 1.57, sorry for the long delay :(
